core/src/nodes/handled_node_tasks.rs

```diff
@@ -450,3 +451,380 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    mod task {
```

Comment:
We shouldn't access anything private in the API. The tests should test the external surface.
I think we don't need these three tests in this module.